Tests/MapboxNavigationTests/StatusViewTests.swift

```diff
@@ -0,0 +1,84 @@
+import XCTest
+import UIKit
+@testable import MapboxNavigation
+@testable import MapboxCoreNavigation
+
+class StatusViewTests: XCTestCase {
+    lazy var statusView: StatusView = {
+        let view: StatusView = .forAutoLayout()
+        view.isHidden = true
+        return view
+    }()
+
+    func testWithDelayShorterThanDuration() {
+        addNewStatus(status: firstStatus())
+        XCTAssertEqual(self.statusView.statuses.count, 1)
+    }
+
+    func testWithDelayLongerThanDuration() {
+        let seconds = 5.0
+        XCTAssertTrue(statusView.isHidden)
+        addNewStatus(status: firstStatus())
+        XCTAssertFalse(statusView.isHidden)
+        let path = #keyPath(UIView.isHidden)
+        let expectation = XCTKVOExpectation(keyPath: path, object: statusView, expectedValue: true)
+        self.wait(for: [expectation], timeout: seconds)
+        XCTAssertTrue(statusView.isHidden)
+        XCTAssertEqual(self.statusView.statuses.count, 0)
+    }
+
+    func testFirstAndSecond() {
+        addNewStatus(status: firstStatus())
+        XCTAssertFalse(statusView.isHidden)
+        addNewStatus(status: secondStatus())
+        XCTAssertEqual(statusView.statuses.count, 2)
+        let path = #keyPath(UIView.isHidden)
+        let expectation = XCTKVOExpectation(keyPath: path, object: statusView, expectedValue: true)
+        self.wait(for: [expectation], timeout: secondStatus().duration + 10.0)
+        XCTAssertTrue(statusView.isHidden)
+        XCTAssertEqual(self.statusView.statuses.count, 0)
+    }
+    
+    func testWithInfinite() {
+        addNewStatus(status: firstStatus())
+        addNewStatus(status: thirdStatus())
+        XCTAssertEqual(self.statusView.statuses.count, 2)
+        let path = #keyPath(UIView.isHidden)
+        let expectation = XCTKVOExpectation(keyPath: path, object: statusView, expectedValue: true)
+        XCTWaiter.wait(for: [expectation], timeout: 15.0)
+        XCTAssertFalse(statusView.isHidden)
+        XCTAssertEqual(self.statusView.statuses.count, 1)
+    }
+}
+
+extension StatusViewTests {
+    
+    // define statuses
+    func firstStatus() -> StatusView.Status {
+        let title1 = NSLocalizedString("FIRST_TEST_STATUS", bundle: .mapboxNavigation, value: "first test status", comment: "the first status banner used for testing")
```

Comment:
There’s more to [adding a localizable string](https://github.com/mapbox/mapbox-navigation-ios/blob/main/CONTRIBUTING.md#adding-user-facing-text) to this codebase, but for an automated test, we actually want the string to be more predictable, so it would be better to avoid `NSLocalizedString` and hard-code `first test status` as is.